GameFinder_WebAPInum2/Services/GameService.cs

```diff
@@ -0,0 +1,144 @@
+﻿using GameFinder_WebAPI.Database;
+using GameFinder_WebAPI.Models;
+using GameFinder_WebAPI.Services.IServices;
+using Microsoft.EntityFrameworkCore;
+
+namespace GameFinder_WebAPI.Services
+{
+    public class GameService : IGameService
+    {
+        private readonly GameDb db;
+
+        public GameService(GameDb db)
+        {
+            this.db = db;
+        }
+
+        public void StartDb()
+        {
+            //database da implementare ogni volta che si avvia il debug
+        }
+
+        public async Task<(ArgumentException?, Game?)> AddGameAsync(Game NewGame)
+        {
+            try
+            {
+                if(NewGame == null)
+                    throw new ArgumentNullException();
+
+                List<Game> games = await db.Games.ToListAsync();
```

Comment:
vale qui come vale in altri posti, non chiamare questo per avere tutti i giochi ma usa .Any, .First, .Where per ottenere al minimo quello che ti serve